staging/src/k8s.io/apiserver/pkg/storage/cacher/delegator/interface.go

```diff
@@ -17,12 +17,67 @@ limitations under the License.
 package delegator
 
 import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apiserver/pkg/features"
 	"k8s.io/apiserver/pkg/storage"
 	etcdfeature "k8s.io/apiserver/pkg/storage/feature"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
 )
 
+func ShouldDelegateList(opts storage.ListOptions, cache Helper) (Result, error) {
+	return shouldDelegateList(listOptions{
+		ResourceVersionMatch: opts.ResourceVersionMatch,
+		ResourceVersion:      opts.ResourceVersion,
+		Continue:             opts.Predicate.Continue,
+		Limit:                opts.Predicate.Limit,
+		Recursive:            opts.Recursive,
+	}, cache)
+}
+
+func ShouldDelegateListMeta(opts *metav1.ListOptions, cache Helper) (Result, error) {
+	return shouldDelegateList(listOptions{
+		ResourceVersionMatch: opts.ResourceVersionMatch,
+		ResourceVersion:      opts.ResourceVersion,
+		Continue:             opts.Continue,
+		Limit:                opts.Limit,
+		Recursive:            true,
+	}, cache)
+}
+
+func shouldDelegateList(opts listOptions, cache Helper) (Result, error) {
+	// see https://kubernetes.io/docs/reference/using-api/api-concepts/#semantics-for-get-and-list
+	switch opts.ResourceVersionMatch {
+	case metav1.ResourceVersionMatchExact:
+		return cache.ShouldDelegateExactRV(opts.ResourceVersion, opts.Recursive)
+	case metav1.ResourceVersionMatchNotOlderThan:
+		return Result{ShouldDelegate: false}, nil
+	case "":
+		// Continue
+		if len(opts.Continue) > 0 {
+			return cache.ShouldDelegateContinue(opts.Continue, opts.Recursive)
+		}
+		// Legacy exact match
+		if opts.Limit > 0 && len(opts.ResourceVersion) > 0 && opts.ResourceVersion != "0" {
+			return cache.ShouldDelegateExactRV(opts.ResourceVersion, opts.Recursive)
+		}
+		// Consistent Read
+		if opts.ResourceVersion == "" {
+			return cache.ShouldDelegateConsistentRead()
+		}
+		return Result{ShouldDelegate: false}, nil
+	default:
+		return Result{ShouldDelegate: true}, nil
+	}
+}
+
+type listOptions struct {
```

Comment:
I wouldn't introduce a new type and just rely on storage.ListOptions

[then the only thing you need is translation when called from APF]

Reply:
Was worried that I might miss some field, or we will add new field and forget to add it to conversion. 
I'm ok with storage options.